File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        visited = {}
        l = 0
        r = 0
        slen = 0

        while r < len(s):
            if s[r] in visited:
                l = max(visited[s[r]]+1,l)
            visited[s[r]] = r
            slen = max(slen, r-l+1)
            print(l,r,s[l],s[r])
            r+=1

        return slen

File: Data Structures & Algorithms/max-water-container/submission-3.java
class Solution {
    public int maxArea(int[] heights) {
        int lp = 0;
        int rp = heights.length-1;
        int resArea = 0;

        while (lp < rp) {
            int area = Math.min(heights[lp], heights[rp]) * (rp - lp);
            resArea = Math.max(resArea, area);

            if(heights[lp] < heights[rp]) {
                lp++;
            } else {
                rp--;
            }
        }
        return resArea;
    }
}
